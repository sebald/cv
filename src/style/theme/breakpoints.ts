export default {
  small: 768,
  medium: 1024,
  large: 1280,
};
