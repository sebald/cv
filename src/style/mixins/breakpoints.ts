import { Interpolation } from 'styled-components';
import { css } from '../styled-components';
import { StyledMixin, Theme, ThemedStyledProps } from '../theme';
import { px } from '../utils';
import { MediaType } from './types';

export const breakpoints: StyledMixin = props => props.theme.breakpoints;

export const media = <P = {}>(
  name: keyof Theme['breakpoints'],
  mediaType: MediaType = 'all'
) => (
  strings: TemplateStringsArray,
  ...interpolations: Interpolation<ThemedStyledProps<P>>[]
) => css`
  @media ${mediaType} and (min-width: ${props =>
      px(breakpoints(props)[name])}) {
    ${css(strings, ...interpolations)};
  }
`;
