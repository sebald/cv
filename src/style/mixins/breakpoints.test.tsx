import { shallow } from 'enzyme';
import * as React from 'react';

import { styled, theme } from '..';
import { breakpoints, media } from './breakpoints';

test('breakpoints', () => {
  expect(breakpoints({ theme })).toMatchSnapshot();
});

test('media', () => {
  const Component = styled('div')`
    ${media('small')`color: red;`};
  `;
  const wrapper = shallow(<Component theme={theme} />);
  expect(wrapper).toMatchSnapshot();
});

test('media with media type', () => {
  const Component = styled('div')`
    ${media('medium', 'screen')`color: red;`};
  `;
  const wrapper = shallow(<Component theme={theme} />);
  expect(wrapper).toMatchSnapshot();
});