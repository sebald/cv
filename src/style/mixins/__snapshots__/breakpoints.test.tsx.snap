// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakpoints 1`] = `
Object {
  "large": 1280,
  "medium": 1024,
  "small": 768,
}
`;

exports[`media 1`] = `
@media all and (min-width:768px) {
  .c0 {
    color: red;
  }
}

<div
  className="c0"
/>
`;

exports[`media with media type 1`] = `
@media screen and (min-width:1024px) {
  .c0 {
    color: red;
  }
}

<div
  className="c0"
/>
`;
